### PR TITLE
Google Cloud Skills Boost README update

### PR DESCRIPTION
**Pull request by asvcode**
updated some portions of the README

## Summary by Sourcery

Enhance the Google Cloud Skills Boost scraper by improving error handling and refactoring the URL handling in the `extract_ml_learning_path` function. Update the README files to provide clearer instructions, including Windows-specific guidance. Modify the `Activity` model to support both integer and float durations.

Enhancements:
- Improve error handling in the `extract_ml_learning_path` function by adding try-except blocks for extracting journey details and links.
- Refactor the `extract_ml_learning_path` function to accept a URL parameter and handle missing data gracefully.
- Add a command-line interface to prompt the user for the GCSB Journey URL and handle CSV writing with error checking.
- Update the `Activity` model to use `Union` for the `duration` field to support both `int` and `float` types.

Documentation:
- Update the README for the Google Cloud Skills Boost scraper to include Windows-specific instructions and correct module paths.
- Add instructions for Windows users to install Microsoft Visual C++ 14.0 or greater in the LLM POC variant README.

### CODE REVIEW

**Review by TobeTek — APPROVED** [review]

**Review by neomatrix369 — COMMENTED** [review]
Hey @asvcode this looks like similar to the PRs you created earlier, do you mind going thru the previous ones one by one and then resolving them, merging them and the rebasing all your branches

I or @TobeTek are happy to walk thru with you on this?

If it's easier to do this over a call, then please let us know

File: app/course-scraper/src/scrapers/google_cloud_skill_boost/scrape_journey.py [hunk 1]
```diff
@@ -17,37 +17,57 @@
 
 
 # Open Journey Path
-def extract_ml_learning_path() -> list[dict]:
+def extract_ml_learning_path(GCSB_JOURNEY_URL) -> list[dict]:
```
[the remark this answers is not in the document]
Comment (neomatrix369) [reply app/course-scraper/src/scrapers/google_cloud_skill_boost/scrape_journey.py:20]: @asvcode is it simple to try this above suggestion? Sourcery has detected something in your implementation and is making these hints

**Comment by asvcode** [comment]
@neomatrix369 @TobeTek Ok I may need a walk through with this, I had created 1 branch and within that branch various commits to the various changes but I see now this is confusing because they are showing up within all the PRs I have submitted.  Can one of you please let me know how best to resolve this.  Each PR should be only the 1 change but because it is on the same branch it is showing up as duplicate. Thanks

**Comment by TobeTek** [comment]
Hi @asvcode , went ahead to restructure these PRs into #107  and #106 . Here's how I achieved that:

1. **Checkout the Pull Request branch:**
Since I don't have a local clone of your fork, I used this command to retrieve the changes from the PR
```bash
git fetch origin pull/82/head:gcsb_readme_update
```

2. **Interactive Rebase to pick desired commits**
Using the rebase keywords and dialog we can pick the commits we'd like and rewrite the commit history, viz:
```bash
git rebase -i d543b98~1  (Brief guide on rebasing)
```

I repeated this for each new PR I created.

**Comment by asvcode** [comment]
@TobeTek Perfect! Thank you for that!

**Review by neomatrix369 — APPROVED** [review]
LGTM

**Pull request merged**